src/main/java/software/amazon/cloudwatchlogs/emf/model/HistogramMetric.java

```diff
@@ -0,0 +1,115 @@
+
+    HistogramMetric(
+            Unit unit,
+            StorageResolution storageResolution,
+            List<Double> values,
+            List<Integer> counts)
+            throws IllegalArgumentException {
+        this(unit, storageResolution, new Histogram(values, counts));
+    }
+
+    protected HistogramMetric(
+            String name, Unit unit, StorageResolution storageResolution, Histogram histogram) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = histogram;
+        this.name = name;
+    }
+
+    HistogramMetric(Unit unit, StorageResolution storageResolution, Histogram histogram) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = histogram;
+    }
+
+    @Override
+    protected LinkedList<Metric> serialize() throws InvalidMetricException {
+        // Histograms will be rejected from CWL if they have more than
+        // Constants.MAX_DATAPOINTS_PER_METRIC number of bins. Unlike MetricDefinition histograms
+        // cannot be broken into multiple messages therefore an error is raised to let users know
+        // their message won't be sent otherwise only this metric will be sent
+        if (isOversized()) {
+            throw new InvalidMetricException(
+                    String.format(
+                            "Histogram metric, %s, has %d values which exceeds the maximum amount "
+                                    + "of bins allowed, %d, and Histograms cannot be broken into "
+                                    + "multiple metrics therefore it will not be published",
+                            name, values.values.size(), Constants.MAX_DATAPOINTS_PER_METRIC));
+        }
+        LinkedList<Metric> metrics = new LinkedList<>();
+        metrics.offer(this);
+        return metrics;
+    }
+
+    @Override
+    protected boolean isOversized() {
+        return values.values.size() > Constants.MAX_DATAPOINTS_PER_METRIC;
+    }
+
+    @Override
+    public boolean hasValidValues() {
+        return values != null && values.count > 0 && !isOversized();
+    }
+
+    public static HistogramMetricBuilder builder() {
+        return new HistogramMetricBuilder();
+    }
+
+    public static class HistogramMetricBuilder
+            extends Metric.MetricBuilder<Histogram, HistogramMetricBuilder> {
+
+        @Override
+        protected HistogramMetricBuilder getThis() {
+            return this;
+        }
+
+        public HistogramMetricBuilder() {
+            this.values = new Histogram();
+        }
+
+        @Override
+        public Histogram getValues() {
+            return values.reduce();
+        }
+
+        @Override
+        public HistogramMetricBuilder addValue(double value) {
+            this.values.addValue(value);
+            return this;
+        }
+
+        @Override
+        public HistogramMetric build() {
+            values.reduce();
+            if (name == null) {
```

Comment:
This condition always returns `false`.

Reply:
the name can be set using setName from the super class metric, which is a protected method